packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -886,12 +891,15 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     const newValue = newProps.value;
     workInProgress.memoizedProps = newProps;
 
-    if (__DEV__) {
+    const providerPropTypes = workInProgress.type.propTypes;
+
+    if (__DEV__ && providerPropTypes) {
```

Comment:
While this is not very important could you please write this as two nested if conditions? We try to make it easy to scan for all DEV conditions in the file so we keep them separately.